CMakeLists.txt

```diff
@@ -22,7 +22,9 @@ include(cmake/utils/utils.cmake)
 
 knowhere_option(WITH_RAFT "Build with RAFT indexes" OFF)
 if (WITH_RAFT)
-  set(CMAKE_CUDA_ARCHITECTURES RAPIDS)
+  if("${CMAKE_CUDA_ARCHITECTURES}" STREQUAL "")
```

Comment:
User should be able to override this if desired. This is particularly useful for setting a value of `NATIVE`, which builds strictly for the local architecture.